### PR TITLE
doc: Set theme in conf.py

### PR DESCRIPTION
**Pull request by aaronmarkham**
*Description of changes:*
The theme of the docs is the `sphinx_rtd_theme`, so it should be set that way in the project, so that when contributors make the docs, they can see the output as it is on production.

Using the env var that was in there adds unnecessary complexity to the setup.

Added info in the readme so setup is repeatable for any contributor.

*Testing done:*
Installed and configured as described here:
https://sphinx-rtd-theme.readthedocs.io/en/stable/installing.html

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -161,14 +161,40 @@ You can also run them in parallel:
...
+    conda create -n sagemaker python=3
+    conda activate sagemaker
+    conda install sphinx
+    pip install sagemaker --user
+
+Install the Read The Docs theme:
+
+::
+
+    pip install sphinx_rtd_theme --user
```
Comment (laurenyu) [inline README.rst]: would it be worth adding a "doc" entry to https://github.com/aws/sagemaker-python-sdk/blob/master/setup.py#L47-L55?

Comment (aaronmarkham) [reply README.rst]: > would it be worth adding a "doc" entry to https://github.com/aws/sagemaker-python-sdk/blob/master/setup.py#L47-L55?

Mmmm no. IDK think it's a good idea to pop installing Sphinx on people. It has a lot of issues with versions and dependency incompatibilities. You could hose someone's setup by pushing a newer version of Sphinx and it's not critical to the core functionality of the sdk.

Comment (laurenyu) [reply README.rst]: neither `sphinx` nor `sphinx_rtd_theme` is in `setup.py` right now for what it's worth. @nadiaya do you have thoughts on this?

Comment (nadiaya) [reply README.rst]: I agree with Aaron, sphinx is not essential to any part of py-sdk functionality.

**Comment by laurenyu** [comment]
the unit test build is failing because flake8 is reporting that `os` is imported but unused in `doc/conf.py`

**Comment by aaronmarkham** [comment]
@laurenyu I can't see why this is failing. Looks like I don't have permissions to view the CodeBuild output. Can you?

**Comment by knakad** [comment]
@aaronmarkham 

We actually have a bot that posts the logs publicly. This allows contributors to view the failure logs.
image

For this particular failure, it looks like a flake8 failure:
```
flake8 run-test: commands[0] | flake8
./doc/conf.py:16:1: F401 'os' imported but unused
```

I'd recommend running `tox tests/unit` (per the contributing readme) to expose the exact failure reason locally =)

**Pull request merged**